pkg/apis/servicecatalog/types.go

```diff
@@ -398,23 +398,30 @@ type ServiceInstance struct {
 	Status ServiceInstanceStatus
 }
 
-// ServiceInstanceSpec represents the desired state of an Instance.
-type ServiceInstanceSpec struct {
-	// ExternalClusterServiceClassName is the human-readable name of the service
-	// as reported by the broker. Note that if the broker changes
+// DesiredPlan defines the user specification for the desired
+// ServicePlan and ServiceClass. Because there are multiple ways to
+// specify the desired Class/Plan, this structure specifies the
+// allowed ways to specify the intent.
+type DesiredPlan struct {
```

Comment:
`DesiredPlanReference`? It's the reference, not the actual plan.
Also I am not sure if "Desired" is a good name for it. More like "Implied", "Presumed" or whatever else makes sense for native English speakers :)

Reply:
Yeah, I too will defer to the native English speakers here :) Since everything eventually boils down to a plan, because each plan belongs to a service class, didn't want to be redundant with adding that there.
PlanReference? :)

Reply:
Yeah, plan reference sounds good.

Reply:
"PlanReference" seems a bit confusing since its not a real K8s ref - which the other "Ref" properties in the Spec will be. This "ref" could technically be a bad "ref" if things get renamed, and having multiple "refs" at the same level for the same thing will be even more confusing.

I actually don't think we should have this wrapper at all since it serves no purpose, other than to force more typing by the end user. The entire ServiceInstanceSpec struct itself is nothing more than information about what service/plan to use, so putting the service/plan in an additional wrapper seems really odd to me. Adding the wrapper doesn't make it any easier to add more types of refs later.